proxy/ControlBase.cc

```diff
@@ -796,58 +795,119 @@ ControlBase::ProcessModifiers(matcher_line *line_info)
   _mods.clear();
   _mods.reserve(n_elts);
 
-  // As elements are consumed, the labels are nulled out and the element
-  // count decremented. So we have to scan the entire array to be sure of
-  // finding all the elements. We'll track the element count so we can
-  // escape if we've found all of the elements.
-  for (int i = 0; n_elts && ME_UNKNOWN == err && i < MATCHER_MAX_TOKENS; ++i) {
-    Modifier *mod = nullptr;
-
-    char *label = line_info->line[0][i];
-    char *value = line_info->line[1][i];
-
-    if (!label) {
-      continue; // Already use.
+  if (!line_info->config_is_yaml) {
+    // As elements are consumed, the labels are nulled out and the element
+    // count decremented. So we have to scan the entire array to be sure of
+    // finding all the elements. We'll track the element count so we can
+    // escape if we've found all of the elements.
+    for (int i = 0; n_elts && ME_UNKNOWN == err && i < MATCHER_MAX_TOKENS; ++i) {
+      Modifier *mod = nullptr;
+
+      char *label = line_info->line[0][i];
+      char *value = line_info->line[1][i];
```

Comment:
Nitpick:  I think it makes the code easier to comprehend if pointers are const when what they point to won't be modified.

Reply:
@ywkaras I think this is out of scope on this PR.  This was existing code and shows up in the diff because it was reformatted when I added the yaml parsing.  I did try to make these const and found that many more out of scope changes are required so, I'll leave this as is for now.